.github/workflows/build.yml

```diff
@@ -0,0 +1,47 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build TaskChampion Sync-Server
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.77
+      - name: Test
+        run: cargo test
+      - name: Build
+        run: cargo build --release --locked
+      - name: Package current compilation
```

Comment:
I think this will end up creating a lot of artifacts. If I understand GH's pricing, we only get 500MB for these? So, perhaps this is best omitted.

Reply:
That data is artifact, it is not something that lives long term and is automatically deleted. [Here is more info](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts).

Reply:
Ah, great - thanks for that!